### DISCUSSION TITLE
Load collection on a specific machine in a cluster

### DISCUSSION THREAD

**Discussion by karolcinik** [milvus-io/milvus#26034] [Q&A and General discussion]
Hi, I have a question regarding collections in a cluster.

Use case

I have a milvus cluster with n machines. To increase performance we would like to load collection on a specific machine in a cluster.
Is it possible?

Also, does milvus caches collections on machines in a cluster?
if yes, what is the caching mechanism.
if no, how to determine what collection has been loaded on a given machine in a cluster.

**Comment by yhmo** [comment]
https://milvus.io/docs/main_components.md

A Milvus cluster has the following components:
- Root coord
maintains the entire system state, generates unique ids for other components, etc.

- Proxy
receive requests from the client, passes requests to different components, etc.

- Query coord
maintains state of query nodes, schedules tasks for query nodes, etc.

- Query node
load segments into memory for search

- Index coord
maintains state of index nodes, schedules tasks for index nodes, etc.

- Index node
build index for segments

- Data coord
maintains state of data nodes, schedules tasks for data nodes, etc.

- Data node
persist data into storage(S3/MinIO), compacts segments, etc.


Milvus search is in-memory, user needs to call collection.load() before search(). Once collection.load() is called, data of the collection will be held in the memory of query nodes, until the user calls collection.release().

Milvus manages data by segments, each segment has an independent index, a segment can be loaded into any query node.
Normally, collection load operation is processed by internal logic. There is an interface that can change the target query node of a segment, but we don't recommend users change the load behavior by the interface. And not all the sdks implemented this interface.
In pymilvus, this interface is [utility.load_balance()](https://github.com/milvus-io/pymilvus/blob/597f3554a865fb7b8582701aaa920aab16194736/pymilvus/orm/utility.py#L552C7-L552C7). It is more like an admin API.
